### ISSUE TITLE
Difficult to implement in existing project setups.

### ISSUE THREAD

**Issue by svenvandescheur** [aurelia/framework#660]
Improvement/Suggestion:
---
Aurelia is close to impossible to use in a way thats not defined by some skeleton. A better way might be like installing react:

`npm install react react-dom`

```js
// Index.js
import React from 'react';
import ReactDOM from 'react-dom';

ReactDOM.render(
  <h1>Hello, world!</h1>,
  document.getElementById('root')
);
```

This allows for a very easy installation and allows a proper way of creating a single bundle (because everything gets explicitly imported, pointing to index.js will do the trick).

For aurelia, its actually quite hard to find a download link, once available. we get presented with ready to go projects. This sound great but is usually is not very practical as many people/organizations already have a lot of build tooling in place (which most of the time doesn't play nice with Aurelia).

For some scenarios an attempt is made to [document](https://github.com/aurelia/framework/blob/master/doc/article/drafts/manual-webpack-configuration.md) an alternative configuration (like for webpack/easy-webpack). This however doesn't seem to completely solve all issues and in my case seems broken (No test coverage in the "vanilla" webpack configuration).

I think aurelia should provide a standalone package that can be explicitly imported into a file from where aurelia gets launched. Currently Aurelia seems to depend to much on third-pary tools which seem to change all the time (jspm/webpack, esnext specification).

Additional questions:
---
- Is it possible to `npm install aurelia`, `import Aurelia from 'aurelia';` and get started without any bundler overhead?
- Are there any plans to simplify the project setup to a bare minimum?
- Are there any plans to make Aurelia less dependent on third party tooling?

**Comment by AshleyGrant** [comment]
We offer a version of the framework that can be loaded using a script tag. You're still going to need a module loader. The script tag version of the framework uses SystemJS as its module loader to support on-the-fly transpilation of code. 

That being said, the React code you posted has the exact same type tooling requirements that you're bemoaning Aurelia for requiring, namely a build pipeline and a module loader. 

You can use the `aurelia.enhance` method to do something like what you want to do. In fact, I have been working on doing something like this for some stuff I've been working on recently. I'm doing mine without any transpiler, and I'm registering modules directly with the module loader, so it is ugly, but you can check it out here: https://github.com/AshleyGrant/aurelia-webforms-testing/blob/master/Script-Tag-Version/index.html

As far as build tooling not playing nicely with Aurelia, we use gulp or webpack in our skeletons. The CLI uses gulp. I can understand how the CLI wouldn't necessarily play nice, but gulp and webpack are both industry standard tooling.

Regarding making Aurelia less dependent on third party tooling, that's why we are working on our CLI.

In short, there's no reason you can't do exactly what you're asking today with Aurelia, it just hasn't been a focus for the team so far, as we haven't had many requests for guidance on this front.

**Comment by svenvandescheur** [comment]
Hi @AshleyGrant 

Thanks for your reply.

The react example indeed needs a
- Module loader/bundler

AFAIK it strictly does not need a build pipeline as React is pretranspiled/packaged. Any loader will allow it to work. Only if you want to use your own ES6 you need "something" to compile that. But that's beyond the scope of react (and I think it should be). It does however document an example config with babel through webpack but we've also managed to set it up with jspm quite easily, anything that can "import" react works.

Wat I'm looking for is not to a script tag/systemjs config or a predefined project structor but just a simple npm module that can be imported using requirejs/imports. To stay with the react analogy this would look something like:

`npm install auralia`

// Index.js
import Aurelia from 'aurelia';

let aurelia = new Aurelia();
aurelia.start();
```

This would allow the dev to
- First setup the plumbing (module loading/project structure/entrypoints etc)
- Then load required dependencies per file.

In order to achieve this Aurelia should not bootstrap using a bundler but should wait to be imported. This might make bootstrapping a little bit harder but to me it seems like a sane thing to do. 

In my opinion the systemjs config/webpack config is the domain of the dev, not of the framework used by the dev.

**Comment by AshleyGrant** [comment]
>In my opinion the systemjs config/webpack config is the domain of the dev, not of the framework used by the dev.

Regarding this, Aurelia doesn't deal with this configuration, it *is* up to the dev and the tooling the dev chooses to configure the module loader.

Now, regarding the code sample you provided. You can do exactly this, and in fact it is pretty much what I do in the code I linked above, except that I use `System.import` instead of an `import` statment and I use the default Aurelia bootstrapper. That being said, you have to understand that simply having an `import` statement (or `require` call) means you will have to configure some module loader. Aurelia itself is not dependent upon any specific module loader, as it works through an abstraction layer. 

You can check out the default bootstrapper here: https://github.com/aurelia/bootstrapper/blob/master/src/index.js to see that it is basically doing what you're asking, but doing it for you. There's NO reason you can't take that file, and rewrite it how you want and use that instead of the default boostrapper we provide. It's really not that much code. It also pretty much does exactly what you're asking.
